src/uhttp.c

```diff
@@ -1053,70 +1059,83 @@ HTTP_CLIENT_RESULT uhttp_client_open(HTTP_CLIENT_HANDLE handle, const char* host
     {
         HTTP_CLIENT_HANDLE_DATA* http_data = (HTTP_CLIENT_HANDLE_DATA*)handle;
 
-        if (http_data->recv_msg.recv_state != state_initial && http_data->recv_msg.recv_state != state_error && http_data->recv_msg.recv_state != state_closed)
+        if ((http_data->recv_msg.recv_state != state_initial) &&
+            (http_data->recv_msg.recv_state != state_error) &&
+            (http_data->recv_msg.recv_state != state_closed))
```

Comment:
>(http_data->recv_msg.recv_state != state_closed) [](start = 12, length = 48)

I'll maybe let someone else with more knowledge about this weigh in but should `state_closing` be added to this list?

Reply:
This was one of the comments from @ewertons : https://github.com/Azure/azure-uhttp-c/issues/48.
We can continue the discussion there.